pkg/process/analyze.go

```diff
@@ -116,10 +122,17 @@ func (a *processAnalyzer) Analyze(pid int, relevantFuncs map[string]interface{})
 	result.GoVersion = goVersion
 	result.Libraries = modules
 
-	start, end := a.findKeyvalMmap(pid)
+	addr, err := a.remoteMmap(pid, mapSize)
+	if err != nil {
+		log.Logger.Error(err, "Failed to mmap")
+		return nil, err
+	}
```

Comment:
Yes, this will cause the instrumentation process to exit.
Running without a successful remote mmap, will cause context propagation to not work properly.
As this new mechanism is much more stable, I think we should treat this as an illegal state and exit.